### DISCUSSION TITLE
How to collect logs into log analytics

### DISCUSSION THREAD

**Discussion by RomanDvorsky** [Azure/api-management-self-hosted-gateway#238] [General]
We would like to collect logs to log analytics workspace or appinsights (does not really matter) however the problem is that we have a lot of namespaces and you can define just which namespace you want to exclude from logging and you cannot define which namespace you want to include in the logging - is there any idea how to collect stderror into any azure resource?

**Comment by tomkerkhove** [comment]
Here's my quick overview:
- If you want to have request logs, you can use our existing Azure Application Insights integration.
- If you are looking to have container logs, then we recommend using regular container monitor tools such as Azure Monitor for Containers.

But I'm adding @ferantoMSFT who is our SHGW PM

**Reply by RomanDvorsky** [reply]
yes but these tools does scrape (ingest) whole cluster, right? We are looking for some solution where we can scrape just one namespace or just specific containers.
Our services on the same cluster does log in appisights directly and we do not want to pay twice for the same logs.

**Reply by tomkerkhove** [reply]
I'll leave it up to @ferantoMSFT to talk about roadmaps and take this feedback as input!
